src/spell/modules/spell/spell/src/kv/collection.rs

```diff
@@ -43,17 +43,20 @@ pub fn list_pop_string(key: &str) -> StringValue {
         get.bind(1, key)?;
         get.bind(2, key)?;
         let string = read_string(&mut get, 1)?;
-        let list_index = get.read::<i64>(3)?;
 
-        let mut delete = db.prepare(
-            r#"
+        if string.is_some() {
```

Comment:
plz, add an explanation why?

Reply:
`list_pop_string` returns `StringValue`:
```rust
pub struct StringValue {
    pub str: String,
    pub success: bool,
    pub error: String,
    pub absent: bool,
}
```
`Ok(Some("blabla"))` converts to `StringValue { success: true, absent: false }`
`Ok(None)` converts to `StringValue { success: true, absent: true }`
`Err(...)` converts to `StringValue { success: false, error: ... }`

So if we have no elements in the list, select returns no rows `read_string(&mut get, 1)?` on it, it will return `Ok(None)` and it will be perfect. But if we try to read the list index for deletion, the whole function will fail with an error on `.read::<i64>(3)?`, so `list_pop_string` returns an error on the empty list instead of `absent`= true.

Reply:
I see. The query is called inside `read_string`. Maybe it's better to manually inline this `read_string` here since we read the result of the query outside of it, wdyt? It feels it's wrong to leave it like that.